library/ecp_curves.c

```diff
@@ -557,6 +557,22 @@ static const mbedtls_mpi_uint brainpoolP512r1_n[] = {
 };
 #endif /* MBEDTLS_ECP_DP_BP512R1_ENABLED */
 
+#if defined(MBEDTLS_ECP_DP_SECP192R1_ENABLED) ||   \
+    defined(MBEDTLS_ECP_DP_SECP224R1_ENABLED) ||   \
+    defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED) ||   \
+    defined(MBEDTLS_ECP_DP_SECP384R1_ENABLED) ||   \
+    defined(MBEDTLS_ECP_DP_SECP521R1_ENABLED) ||   \
+    defined(MBEDTLS_ECP_DP_BP256R1_ENABLED)   ||   \
+    defined(MBEDTLS_ECP_DP_BP384R1_ENABLED)   ||   \
+    defined(MBEDTLS_ECP_DP_BP512R1_ENABLED)   ||   \
+    defined(MBEDTLS_ECP_DP_SECP192K1_ENABLED) ||   \
+    defined(MBEDTLS_ECP_DP_SECP224K1_ENABLED) ||   \
+    defined(MBEDTLS_ECP_DP_SECP256K1_ENABLED)
```

Comment:
As the comment explains, this is the list of curves for which `ecp_group_load` is necessary.

A curve may be removed from the list if we write an optimized implementation that doesn't have a one-time initialization pase. Conversely a new curve may be added to the list even if it isn't a short Weierstrass curve. It's a matter of how the implementation works.

The order here is the same as in the definition of `mbedtls_ecp_group_id` (which we've kept stable because it's part of the ABI). I guess I should change the order in the definition of `MBEDTLS_ECP_SHORT_WEIERSTRASS_ENABLED` to match.